ha/util/actuator_req_manager.py

```diff
@@ -0,0 +1,246 @@
+        print(f"Read machine-id. Output: {machine_id}")
+        return machine_id.strip()
+
+    def enclosure_stop(self, node_name: str) -> bool :
+        """
+        Send actuator request to monitor for stopping enclosure
+
+        Args:
+            node_name : node on which enclosure is to be stopped
+
+        Return:
+            True : if enclose stop successful; else exception is raised
+        """
+
+        req = self._create_req(node_name)
+        self._send_req(req)
+        self._wait_for_resp()
+
+        # Wait for 60 sec max. Expected max wait time = 40 sec + 20 sec buffer
+        for _ in range(0, ACTUATOR_RESP_RETRY_COUNT):
+            time.sleep(ACTUATOR_RESP_WAIT_TIME)
+            if self._is_resp_received:
+               break
+
+        # Stop the thread that was created
+        # TBD once the changes are done as part of PR#530.
+        #self.consumer.stop()
+
+        if self._is_resp_received and self._encl_shutdown_successful:
+            Log.info(f"Enclosure shutdown successful on node {node_name}")
+            self._is_resp_received = self._encl_shutdown_successful = False
+            return True
+
+        if not self._is_resp_received:
+            Log.error(f"Actuator response not received; enclosure shutdown failed on node {node_name}")
+        else:
+            Log.error(f"Unable to shutdown enclosure on node {node_name}")
+
+        self._is_resp_received = self._encl_shutdown_successful = False
+        raise ClusterManagerError("Failed to shutdown the enclosure")
+
+    def _create_req(self, target_node_name: str) -> str:
+        """
+        Create actualtor request for enclosure stop
+
+        Args:
+            target_node_name : node on which enclosure is to be stopped
+
+        Return:
+            created actuator request
+        """
+
+        with open(ACTUATOR_SCHEMA, 'r') as actuator_req_schema_file:
+            actuator_req = json.load(actuator_req_schema_file)
+
+        # Get details for current node
+        node_name = Conf.get(const.HA_GLOBAL_INDEX, f"CLUSTER_MANAGER{_DELIM}local_node")
+
+        key_val  = self._conf_store.get(f"{PVTFQDN_TO_NODEID_KEY}/{node_name}")
+        _, node_id = key_val.popitem()
+
+        system_health = SystemHealth(self._conf_store)
+        health_manager = SystemHealthManager(self._conf_store)
+        key = system_health._prepare_key(const.COMPONENTS.NODE_MAP.value, node_id=node_id)
+
+        node_map_val = health_manager.get_key(key)
+        if node_map_val is None:
+            raise ClusterManagerError("Failed to fetch node_map value")
+
+        node_map_dict = ast.literal_eval(node_map_val)
+        self._generate_uuid()
+
+        # Populate the actuator request schema
+        actuator_req[ACTUATOR_ATTRIBUTES.TIME] = str(int(time.time()))
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.REQUEST_PATH][ACTUATOR_ATTRIBUTES.SITE_ID] = node_map_dict[NODE_MAP_ATTRIBUTES.SITE_ID.value]
```

Comment:
Please confirm with "monitor" team that these details are for the node sending the request or for the target node.

Reply:
This is already confirmed